functions/Get-DbaLastGoodCheckDb.ps1

```diff
@@ -0,0 +1,183 @@
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+Data Purity:
+		http://www.sqlskills.com/blogs/paul/checkdb-from-every-angle-how-to-tell-if-data-purity-checks-will-be-run/
+		https://www.mssqltips.com/sqlservertip/1988/ensure-sql-server-data-purity-checks-are-performed/
+
+.EXAMPLE
+Get-DbaLastGoodCheckDb -SqlServer ServerA\sql987
+
+Server         Database      LastGoodCheckDb
+------         --------      ---------------
+ServerA\sql987 badDB         11-06-2014 20:17:29
+ServerA\sql987 databaselist  11-06-2014 20:17:42
+...
+
+.EXAMPLE
+Get-DbaLastGoodCheckDb -SqlServer ServerA\sql987 -Detailed | Format-Table -AutoSize
+
+Server         Database      DatabaseCreated     LastGoodCheckDb     DaysSinceDbCreated DaysSinceLastGoodCheckDb Status                      DataPurityEnabled
+------         --------      ---------------     ---------------     ------------------ ------------------------ ------                      -----------------
+ServerA\sql987 badDB         14-10-2013 09:21:49 11-06-2014 20:17:29               1078                      838 CheckDB should be performed False
+ServerA\sql987 databaselist  23-10-2013 17:31:35 11-06-2014 20:17:42               1069                      838 CheckDB should be performed True
+...
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory = $true, ValueFromPipeline = $true)]
+		[Alias("ServerInstance", "SqlInstance")]
+		[string[]]$SqlServer,
+		[PsCredential]$Credential,
+		[switch]$Detailed
+	)
+
+	DynamicParam {
+		if ($SqlServer) {
+			return Get-ParamSqlDatabases -SqlServer $SqlServer[0] -SqlCredential $Credential
+		}
+	}
+
+	BEGIN
+	{
+		# Convert from RuntimeDefinedParameter object to regular array
+		$databases = $psboundparameters.Databases
+		$exclude = $psboundparameters.Exclude
+
+		$collection = New-Object System.Collections.ArrayList
+	}
+
+	PROCESS
+	{
+		foreach ($servername in $SqlServer)
+		{
+			try
+			{
+				$server = Connect-SqlServer -SqlServer $servername -SqlCredential $Credential
+			}
+			catch
+			{
+				if ($SqlServer.count -eq 1)
+				{
+					throw $_
+				}
+				else
+				{
+					Write-Warning "Can't connect to $servername. Moving on."
+					Continue
+				}
+			}
+
+			$dbs = $server.Databases
+
+			if ($databases.count -gt 0)
+			{
+				$dbs = $dbs | Where-Object { $databases -contains $_.Name }
+			}
+
+			if ($exclude.count -gt 0)
+			{
+				$dbs = $dbs | Where-Object { $exclude -notcontains $_.Name }
+			}
+
+			foreach ($db in $dbs)
+			{
+				Write-Verbose "Processing $($db.name) on $servername"
+
+				$sql = "DBCC DBINFO ([$($db.name)]) WITH TABLERESULTS"
+				Write-Verbose "T-SQL: $sql"
+
+				$resultTable = $db.ExecuteWithResults($sql).Tables[0]
+				[datetime]$lastKnownGood = ($resultTable | Where-Object Field -eq 'dbi_dbccLastKnownGood').Value
```

Comment:
This has issues for me on some servers. It seems the field dbi_dbccLastKnownGood exists twice on some databases on some of my test servers. Is that normal? Either way, it causes the cmdlet to have issues assigning a value to this variable then causes other errors further down!